lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -307,6 +309,42 @@ sub register {
             my ($c, $group) = @_;
             return $c->tag('li', $c->link_to($group->name => $c->url_for('group_overview', groupid => $group->id)));
         });
+
+    $app->helper(
+        current_version => sub {
+            my $c = shift;
+
+            # If the value is not in the application defaults
+            my $current_version = $c->app->defaults('current_version');
+            unless ($current_version) {
+                my $version;
+                my $dir = abs_path($FindBin::Bin . "/..");
+
+                if (-e $dir . '/.git' and -e $dir . '/.git/refs/heads/master') {
+                    $version = file_content($dir . '/.git/refs/heads/master');
+
+                    # Extrapolate latest tagged version and combine it
+                    # with latest commit which heads is pointed to.
+                    # This method have its limits while checking out different branches
+                    # but emulates git-describe output without executing commands.
+                    my $packed_refs = file_content($dir . '/.git/packed-refs');
+                    if ($packed_refs) {
+                        my $latest_ref = (grep(/tags/, split(/\s/, $packed_refs)))[-1];
```

Comment:
I recommend to shorten the git hash to 8 characters which is still unambiguous (enough)